docs/docs/addons-test-utils.md

```diff
@@ -0,0 +1,323 @@
+
+  - [`createRenderer()`](#createrenderer)
+  - [`render()`](#render)
+  - [`getRenderOutput()`](#getrenderoutput)
+
+Shallow testing currently has some limitations, namely not supporting refs. We're releasing this feature early and would appreciate the React community's feedback on how it should evolve.
+
+* * *
+
+## Reference
+
+### `Simulate`
+
+```javascript
+Simulate.{eventName}(
+  element,
+  [eventData]
+)
+```
+
+Simulate an event dispatch on a DOM node with optional `eventData` event data.
+
+`Simulate` has a method for [every event that React understands](/react/docs/events.html#supported-events).
+
+**Clicking an element**
+
+```javascript
+// <button ref="button">...</button>
+var node = this.refs.button;
+ReactTestUtils.Simulate.click(node);
+```
+
+**Changing the value of an input field and then pressing ENTER.**
+
+```javascript
+// <input ref="input" />
+var node = this.refs.input;
+node.value = 'giraffe';
+ReactTestUtils.Simulate.change(node);
+ReactTestUtils.Simulate.keyDown(node, {key: "Enter", keyCode: 13, which: 13});
+```
+
+> Note
+>
+> You will have to provide any event property that you're using in your component (e.g. keyCode, which, etc...) as React is not creating any of these for you.
+
+* * *
+
+### `renderIntoDocument()`
+
+```javascript
+renderIntoDocument(instance)
+```
+
+Render a component into a detached DOM node in the document. **This function requires a DOM.**
+
+> Note:
+>
+> You will need to have `window`, `window.document` and `window.document.createElement` globally available **before** you import `React`. Otherwise React will think it can't access the DOM and methods like `setState` won't work.
+
+* * *
+
+### `mockComponent()`
+
+```javascript
+mockComponent(
+  componentClass,
+  [mockTagName]
+)
+```
+
+Pass a mocked component module to this method to augment it with useful methods that allow it to be used as a dummy React component. Instead of rendering as usual, the component will become a simple `<div>` (or other tag if `mockTagName` is provided) containing any provided children.
+
+* * *
+
+### `isElement()`
+
+```javascript
+isElement(element)
+```
+
+Returns `true` if `element` is any React element.
+
+* * *
+
+### `isElementOfType()`
+
+```javascript
+isElementOfType(
+  element,
+  componentClass
+)
+```
+
+Returns `true` if `element` is a React element whose type is of a React `componentClass`.
+
+* * *
+
+### `isDOMComponent()`
+
+```javascript
+isDOMComponent(instance)
+```
+
+Returns `true` if `instance` is a DOM component (such as a `<div>` or `<span>`).
+
+* * *
+
+### `isCompositeComponent()`
+
+```javascript
+isCompositeComponent(instance)
+```
+
+Returns `true` if `instance` is a composite component (a `React.Component` subclass or a component created with `React.createClass()`).
```

Comment:
Or a function. I would avoid this list of things here and just say "if instance is a user-defined component, such as a class or a function".